src/commands/wallet/balance.rs

```diff
@@ -3,39 +3,75 @@ use clap::Args as ClapArgs;
 use eyre::ContextCompat;
 use std::sync::Arc;
 use zksync_ethers_rs::{
-    contracts::erc20::ERC20,
+    contracts::{erc20::ERC20, l2_shared_bridge::get_l2_token_from_l1_address},
+    core::utils::{format_ether, format_units},
     providers::{Middleware, Provider},
     types::Address,
+    utils::L2_ETH_TOKEN_ADDRESS,
+    ZKMiddleware,
 };
 
 #[derive(ClapArgs)]
 pub(crate) struct Args {
     #[clap(long = "token")]
     pub token_address: Option<Address>,
+    #[clap(long = "tol2", required = false)]
+    pub l2_token: bool,
     #[clap(long = "l1", required = false)]
     pub l1: bool,
 }
 
 pub(crate) async fn run(args: Args, cfg: ZKSyncConfig) -> eyre::Result<()> {
+    let l1_provider = Provider::try_from(
+        cfg.network
+            .l1_rpc_url
+            .context("L1 RPC URL missing in config")?,
+    )?;
+    let l2_provider = Provider::try_from(cfg.network.l2_rpc_url)?;
```

Comment:
I'd avoid instantiating a provider for each network in every use case of the command because this will require the user to configure the L1 RPC even if they just need to query L2 balances.

We can instantiate the L1 provider only when `--l1` is set.

Reply:
Yes, but we are using the `l1_provider` to get the `token_symbol` of the `base_token`.